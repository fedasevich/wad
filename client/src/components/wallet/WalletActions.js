import { observer } from 'mobx-react-lite';
import React from 'react';
import { Col, Row } from 'react-bootstrap';
import { EditIcon, RechargeIcon, TransactionsIcon, TransferIcon } from '../../ui/Icons/WalletIcons/WalletIcons';
import { WALLET_PAGE_STATE } from '../../utils/constants';

export const WALLET_ACTIONS_AND_ICONS = [
  { id: 1, icon: <EditIcon />, action: WALLET_PAGE_STATE.EDIT_WALLET, name: 'Edit' },
  { id: 2, icon: <RechargeIcon />, action: WALLET_PAGE_STATE.RECHARGE_WALLET, name: 'Recharge' },
  { id: 3, icon: <TransactionsIcon />, action: WALLET_PAGE_STATE.TRANSACTIONS_WALLET, name: 'Transactions' },
  { id: 4, icon: <TransferIcon />, action: WALLET_PAGE_STATE.TRANSFER_WALLET, name: 'Transfer' }
];

const WalletActions = observer(({ dispatch, id }) => {
  return (
    <Row>
      {WALLET_ACTIONS_AND_ICONS.map((walletActionAndIcon) => (
        <Col
          sm="6"
          md="6"
          className="mb-3 px-xl-2 d-flex justify-content-xl-center justify-content-xxl-start"
          key={walletActionAndIcon.id}
        >
          <button
            type="button"
            className="cursor-pointer d-flex align-items-center flex-lg-row flex-xl-column flex-xxl-row justify-content-center text-center"
            onClick={() => {
              dispatch({ operation: walletActionAndIcon.action, id });
            }}
          >
            {walletActionAndIcon.icon}
            <span className="ms-2 ms-md-2 ms-xl-0 ms-xxl-2">{walletActionAndIcon.name}</span>
          </button>
        </Col>
      ))}
    </Row>
  );
});

export default WalletActions;
