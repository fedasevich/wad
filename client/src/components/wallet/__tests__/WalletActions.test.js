import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import WalletActions, { WALLET_ACTIONS_AND_ICONS } from '../WalletActions';

jest.mock('mobx-react-lite', () => ({
  observer: jest.fn((component) => component)
}));

describe('WalletActions.js', () => {
  it('renders all wallet action buttons correctly', () => {
    render(<WalletActions dispatch={() => undefined} id={1} />);
    const actionButtons = screen.getAllByRole('button');

    expect(actionButtons).toHaveLength(WALLET_ACTIONS_AND_ICONS.length);
  });

  it('dispatches the correct action when a button is clicked', () => {
    const dispatchMock = jest.fn();
    render(<WalletActions dispatch={dispatchMock} id={1} />);

    WALLET_ACTIONS_AND_ICONS.forEach((actionAndIcon) => {
      const button = screen.getByText(actionAndIcon.name);
      fireEvent.click(button);

      expect(dispatchMock).toHaveBeenCalledWith({
        operation: actionAndIcon.action,
        id: 1
      });
    });
  });

  it('matches snapshot', () => {
    const { container } = render(<WalletActions dispatch={() => undefined} id={1} />);
    expect(container).toMatchSnapshot();
  });
});
