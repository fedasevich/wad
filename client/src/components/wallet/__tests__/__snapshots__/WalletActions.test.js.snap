// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletActions.js matches snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="mb-3 px-xl-2 d-flex justify-content-xl-center justify-content-xxl-start col-md-6 col-sm-6"
    >
      <button
        class="cursor-pointer d-flex align-items-center flex-lg-row flex-xl-column flex-xxl-row justify-content-center text-center"
        type="button"
      >
        <svg>
          Icon-1.svg
        </svg>
        <span
          class="ms-2 ms-md-2 ms-xl-0 ms-xxl-2"
        >
          Edit
        </span>
      </button>
    </div>
    <div
      class="mb-3 px-xl-2 d-flex justify-content-xl-center justify-content-xxl-start col-md-6 col-sm-6"
    >
      <button
        class="cursor-pointer d-flex align-items-center flex-lg-row flex-xl-column flex-xxl-row justify-content-center text-center"
        type="button"
      >
        <svg>
          Icon-3.svg
        </svg>
        <span
          class="ms-2 ms-md-2 ms-xl-0 ms-xxl-2"
        >
          Recharge
        </span>
      </button>
    </div>
    <div
      class="mb-3 px-xl-2 d-flex justify-content-xl-center justify-content-xxl-start col-md-6 col-sm-6"
    >
      <button
        class="cursor-pointer d-flex align-items-center flex-lg-row flex-xl-column flex-xxl-row justify-content-center text-center"
        type="button"
      >
        <svg>
          Icon-2.svg
        </svg>
        <span
          class="ms-2 ms-md-2 ms-xl-0 ms-xxl-2"
        >
          Transactions
        </span>
      </button>
    </div>
    <div
      class="mb-3 px-xl-2 d-flex justify-content-xl-center justify-content-xxl-start col-md-6 col-sm-6"
    >
      <button
        class="cursor-pointer d-flex align-items-center flex-lg-row flex-xl-column flex-xxl-row justify-content-center text-center"
        type="button"
      >
        <svg>
          Icon-5.svg
        </svg>
        <span
          class="ms-2 ms-md-2 ms-xl-0 ms-xxl-2"
        >
          Transfer
        </span>
      </button>
    </div>
  </div>
</div>
`;
